staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/marshal_cbor.go

```diff
@@ -0,0 +1,35 @@
+package v1
+
+import (
+	"github.com/fxamacker/cbor/v2"
+)
+
+var _ cbor.Marshaler = JSONSchemaPropsOrBool{}
+var _ cbor.Unmarshaler = &JSONSchemaPropsOrBool{}
+
+var (
+	cborFalse = []byte{0xf4}
+	cborTrue  = []byte{0xf5}
+)
+
+func (s JSONSchemaPropsOrBool) MarshalCBOR() ([]byte, error) {
+	if s.Schema != nil {
+		return cbor.Marshal(s.Schema)
+	}
+
+	if s.Schema == nil && !s.Allows {
+		return cborFalse, nil
+	}
+	return cborTrue, nil
```

Comment:
I see that it's carried over from the JSON implementation, but isn't the nil check redundant here? Could be:

```suggestion
	return cbor.Marshal(s.Allows)
```